lib/ddtrace/pipeline.rb

```diff
@@ -9,6 +9,14 @@ module Pipeline
     @mutex = Mutex.new
     @processors = []
 
+    # {.before_flush} allows application to alter or filter out traces before they are flushed.
+    #
+    # @see file:docs/ProcessingPipeline.md Processing Pipeline
+    # @param [Array<Proc>] processors a list of callable objects that receive a list of {Datadog::Span}s and can modify
+    #   or filter our spans.
```

Comment:
Not sure how to represent it in the docs, but this is actually not an `Array<Proc>`, since we only want `#call` duck types. Also, "filter our spans" sounds a bit from the library developer POV, I suggest "filter a trace's spans" for a more neutral wording.

Reply:
We can do it like this `Array<#call(Datadog::TraceSegment)>`.

But overall, I've rewritten this doc, it should read better now.